### PR TITLE
Preserve custom ops via run_decomps

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #136888
* #136887
* #136884
* __->__ #136882

This is re-apply of https://github.com/pytorch/pytorch/pull/136773?fbclid=IwZXh0bgNhZW0CMTEAAR3SmginkvZcILVY7G2XDa_KosnV4DPmq1l6pkjPIM255QgJLKVAR90rGAU_aem_ZWpcVdUsmAGzOGiwbjtBDg. 

Note that this doesn't completely remove the _preserve_ops list from export mainly because we want to have small change to address failing executorch tests. All the complications included in this PR is deleted in the next PR. 

Differential Revision: [D63553086](https://our.internmc.facebook.com/intern/diff/D63553086/)

### CODE REVIEW

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D63553086).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D63553086).

File: torch/_decomp/__init__.py [hunk 1]
```diff
@@ -269,12 +269,15 @@ def remove_decompositions(
 #    is mutating or aliasing.
 # TODO (tmanlaibaatar) make this utility function and share it with functional_tensor
 # decomp part. (https://github.com/pytorch/pytorch/issues/129431)
-def _check_valid_to_preserve(op_overload):
+def _check_valid_to_preserve(op_overload: "OperatorBase"):
```
Comment (tugsbayasgalan) [inline torch/_decomp/__init__.py:272]: Previous discussion: https://github.com/pytorch/pytorch/pull/136773#discussion_r1777835890

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D63553086).

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D63553086).

**Review by bdhirsh — APPROVED** [review]
ok, so this PR is explicitly to allow CIA custom ops to be preserved using the `_preserve_ops` kwarg. And later in the stack, I guess the plan is to get the same behavior but without `_preserve_ops`?

**Pull request closed without merging**